i18n-helpers/src/bin/markdown-gettext.rs

```diff
@@ -0,0 +1,248 @@
+use anyhow::{anyhow, bail, Context};
+use mdbook_i18n_helpers::translate;
+use polib::catalog::Catalog;
+use polib::po_file;
+use std::fs::File;
+use std::io::{BufReader, Read, Write};
+use std::path::PathBuf;
+use std::{env, fs};
+
+fn split_argument(arg: &String) -> Option<String> {
+    let splitted: Vec<_> = arg.split('=').collect();
+    if splitted.len() == 2 {
+        return Some(splitted[1].to_string());
+    }
+
+    None
+}
+
+fn build_catalog(lang_option: Option<String>) -> anyhow::Result<Catalog> {
+    match lang_option {
+        Some(lang_path) => {
+            let pot_path = PathBuf::from(&lang_path);
```

Comment:
I believe `PathBuf` is only for when you need a _mutable buffer_. If you just need a read-only path, you should use `Path`.